### ISSUE TITLE
Improve the transfers of DIGIs and clusters in Raw2Cluster

### ISSUE THREAD

**Issue by makortel** [cms-patatrack/cmssw#67]
#### Digis
After #62 the DIGIS+errors are still always transferred back to CPU. In the pixel tracking workflow they are not used, so it should be straightforward do make their transfer on-demand. Basically this requires moving the transfers
https://github.com/cms-patatrack/cmssw/blob/cdcd677b010bad75f67b740350e88261b2b03e1a/EventFilter/SiPixelRawToDigi/plugins/SiPixelRawToClusterGPUKernel.cu#L645-L655
from the kernel to the "transfer function"
https://github.com/cms-patatrack/cmssw/blob/cdcd677b010bad75f67b740350e88261b2b03e1a/EventFilter/SiPixelRawToDigi/plugins/SiPixelRawToClusterHeterogeneous.cc#L517-L518

On a longer term (going beyond the quadruplet pixel tracks), the error information will likely become useful, so one could think of splitting the DIGIs and the errors to two separate `HeterogeneousProduct`s (with the cost of them having different product labels...).

#### Clusters
The clusters, on the other hand, are needed in the CPU for various reasons (RecHits need to make edm::Refs, MC matching relies on clusters etc). After #62 the CPU clusters are first put as part of `HeterogeneousProduct`, and then copied to a standard `edmNew::DetSetVector<SiPixelCluster>` by `SiPixelClusterHeterogeneousConverter`. Since the standard collection is needed anyway, it would be more optimal to produce it directly already by `SiPixelRawToClusterHeterogeneous`. (this will then lead to a funny situation of module `siPixelDigis` taking `siPixelCluters` as an input, but ok)

**Comment by cmsbot** [comment]
A new Issue was created by  @makortel Matti Kortelainen.

 can you please review it and eventually sign/assign? Thanks.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-issues.html">here</a>

**Comment by makortel** [comment]
@felicepantaleo @VinInn

**Comment by makortel** [comment]
With the approach of #100 this issue would get fixed automatically.
